modules/api/src/main/java/net/countercraft/movecraft/craft/Craft.java

```diff
@@ -322,25 +324,25 @@ public int getTickCooldown() {
 
         int chestPenalty = (int)((materials.get(Material.CHEST) + materials.get(Material.TRAPPED_CHEST)) * type.getChestPenalty());
         if(!cruising)
-            return type.getTickCooldown(w) + chestPenalty;
+            return (type.getTickCooldown(w) * currentGear) + chestPenalty;
 
         // Ascent or Descent
         if(cruiseDirection == 0x42 || cruiseDirection == 0x43) {
-            return type.getVertCruiseTickCooldown(w) + chestPenalty;
+            return (type.getVertCruiseTickCooldown(w) * currentGear) + chestPenalty;
         }
 
         // Dynamic Fly Block Speed
         if(type.getDynamicFlyBlockSpeedFactor() != 0){
             Material flyBlockMaterial = Material.getMaterial(type.getDynamicFlyBlock());
             double count = materials.get(flyBlockMaterial);
             double woolRatio = count / hitBox.size();
-            return Math.max((int)Math.round((20.0 * (type.getCruiseSkipBlocks(w) + 1)) / ((type.getDynamicFlyBlockSpeedFactor() * 1.5) * (woolRatio - .5) + (20.0 / type.getCruiseTickCooldown(w)) + 1)), 1);
+            return max((int)Math.round((20.0 * (type.getCruiseSkipBlocks(w) + 1)) / ((type.getDynamicFlyBlockSpeedFactor() * 1.5) * (woolRatio - .5) + (20.0 / (type.getCruiseTickCooldown(w) * currentGear)) + 1)), 1);

```

Comment:
This one will have a ton of fun consequences, best to multiply the results of the max by the gearshift.